packages/slate-commons/src/commands/isEmptyParagraphElement.ts

```diff
@@ -1,28 +1,39 @@
 import { isParagraphNode } from '@prezly/slate-types';
-import { Node } from 'slate';
-import type { Editor } from 'slate';
+import { Editor, Node, Text } from 'slate';
 
-import { isNodeEmpty } from './isNodeEmpty';
-
-export interface IsEmptyParagraphElementOptions {
-    shouldTrim?: boolean;
+interface Options {
+    trim?: boolean;
 }
 
 export function isEmptyParagraphElement(
     editor: Editor,
     node?: Node | null,
-    options?: IsEmptyParagraphElementOptions,
+    options?: Options,
 ): boolean {
     if (!isParagraphNode(node) || !node) {
         return false;
     }
 
-    let nodeEmpty = isNodeEmpty(editor, node);
+    if (hasVoidElements(editor, node)) {
+        return false;
+    }
```

Comment:
@aspirisen I've also realized that we should take into account void nodes (as there can be inline voids too): 58abb816245426d8ba21bddb2351f4d4d0301e74